src/coreclr/utilcode/sigparser.cpp

```diff
@@ -192,3 +192,180 @@ HRESULT SigParser::SkipSignature()
 
     return hr;
 }
+
+HRESULT SigParser::MoveToSignature(uint32_t indexToFind)
+{
+    CONTRACTL
+    {
+        INSTANCE_CHECK;
+        NOTHROW;
+        GC_NOTRIGGER;
+        FORBID_FAULT;
+        SUPPORTS_DAC;
+    }
+    CONTRACTL_END
+
+    BOOL isFinished = FALSE;
+    HRESULT hr = MoveToSignature(indexToFind, 0, &isFinished);
+    if (isFinished == FALSE)
+    {
+        return META_E_BAD_SIGNATURE;
+    }
+
+    return hr;
+}
+
+HRESULT SigParser::MoveToSignature(uint32_t indexToFind, uint32_t currentIndex, BOOL* isFinished)
```

Comment:
This does not seem to match the managed logic for computing the index. For example, this should print `System.Runtime.CompilerServices.CallConvCdecl`, but it prints nothing - because it skips to a wrong function pointer: 
```
using System;
using System.Collections.Generic;
using System.Runtime.CompilerServices;

var mt = typeof(My).GetProperty("MyProp").GetModifiedPropertyType();
var f1 = mt.GetElementType();
var f2 = f1.GetFunctionPointerParameterTypes()[1];
var f3 = f2.GetFunctionPointerParameterTypes()[2];
foreach (var cc in f3.GetFunctionPointerCallingConventions())
    Console.WriteLine(cc.ToString());

public unsafe class My
{
    public delegate* <delegate* <void>, delegate* <delegate* <void>, delegate* <void>, delegate* unmanaged[Cdecl]<void>, void>, void>[] MyProp { get; }
}
```

Would it better to remember the pointer into the signature where we stopped parsing and where to restart? It looks fragile to recreate position within signature using the index by depending on two pieces of logic that are very far apart to be 100% in sync.

Reply:
I did fix this locally. Thanks.

> Would it better to remember the pointer into the signature where we stopped parsing and where to restart?

Each node can be accessed at random, so there isn't really a start\stop. If we could take a reference to S.R.Metadata I probably would have just read the metadata that way and populated the position or metadata for every node in one pass.

Reply:
This was fixed and I added the test above.